src/lib/ui/uiMutationHelper.ts

```diff
@@ -27,3 +31,34 @@ export const upsertCommunityFreshnessById = (ui: WritableUi, community_id: numbe
 		freshnessByCommunityId.set(community_id, writable(fresh));
 	}
 };
+
+export const updateLastSeen = (
+	ui: WritableUi,
+	directory_id: number,
+	time: number | undefined,
+): void => {
+	const {lastSeenByDirectoryId, entityById, spaceById} = ui;
+	const timestamp = time ?? Date.now();
+
+	if (lastSeenByDirectoryId.has(directory_id)) {
+		lastSeenByDirectoryId.get(directory_id)!.set(timestamp);
+	} else {
+		lastSeenByDirectoryId.set(directory_id, writable(timestamp));
+	}
+
+	if (browser) {
+		localStorage.setItem(`${LAST_SEEN_KEY}${directory_id}`, `${timestamp}`);
```

Comment:
looks like it can be simplified a bit:

```suggestion
	time = Date.now(),
): void => {
	const {lastSeenByDirectoryId, entityById, spaceById} = ui;

	if (lastSeenByDirectoryId.has(directory_id)) {
		lastSeenByDirectoryId.get(directory_id)!.set(time);
	} else {
		lastSeenByDirectoryId.set(directory_id, writable(time));
	}

	if (browser) {
		localStorage.setItem(`${LAST_SEEN_KEY}${directory_id}`, `${time}`);
```